osu.Game/Overlays/RankingsOverlay.cs

```diff
@@ -9,31 +9,35 @@
 using osu.Game.Overlays.Rankings;
 using osu.Game.Users;
 using osu.Game.Rulesets;
-using osu.Game.Online.API;
 using System.Threading;
 using osu.Game.Graphics.UserInterface;
+using osu.Game.Overlays.Rankings.Displays;
+using System;
 using osu.Game.Online.API.Requests;
-using osu.Game.Overlays.Rankings.Tables;
+using osu.Game.Online.API;
 
 namespace osu.Game.Overlays
 {
     public class RankingsOverlay : FullscreenOverlay
     {
-        protected Bindable<Country> Country => header.Country;
+        public const float CONTENT_X_MARGIN = 50;
 
         protected Bindable<RankingsScope> Scope => header.Current;
 
-        private readonly OverlayScrollContainer scrollFlow;
+        [Resolved]
+        private Bindable<RulesetInfo> ruleset { get; set; }
+
         private readonly Container contentContainer;
         private readonly LoadingLayer loading;
         private readonly Box background;
         private readonly RankingsOverlayHeader header;
+        private readonly OverlayScrollContainer scrollFlow;
 
-        private APIRequest lastRequest;
+        private GetSpotlightsRequest spotlightsRequest;
         private CancellationTokenSource cancellationToken;
 
-        [Resolved]
-        private IAPIProvider api { get; set; }
+        private PerformanceRankingsDisplay performanceDisplay;
```

Comment:
I'd be more accepting of this if it were of a type that encapsulates all possible displays and pattern-matches to check whether it's a performance display wherever necessary.